src/device-api/common.js

```diff
@@ -30,7 +30,10 @@ export async function doRestart(appId, force) {
 
 			return deviceState.pausingApply(async () => {
 				return Bluebird.each(app.services, async (service) => {
-					await serviceManager.kill(service, { wait: true });
+					await serviceManager.kill(service, {
+						wait: true,
+						removeContainer: false,
```

Comment:
Adding my 2 cents... Is this change really desirable though? And is it effectively a breaking change, when the previous behavior has been around for at least 1.5 years, and maybe since ever? Not that breaking changes cannot be made, but they perhaps warrant a bit more discussion.

On one hand, one could argue that we should try to emulate `docker restart` and that restart should mean stop + start because "that's what it usually means". On the other hand, isn't stop + start "less useful" than re-creating the container? If an app service got stuck in a bad state, recreating the container is more useful and powerful than stop + start, which might simply "resume being stuck". Perhaps if the supervisor / dashboard offered an additional "reset" action as suggested by Joe in the support thread, then this change to restart could be introduced at the same time, and users would still have a way of recreating containers other than using restart.

One could also make the case that the only thing we need is to document that restart means recreating the container, and not changing what it currently does. Check this comment about the dashboard having a tooltip to clarify to users that data saved to the container will be lost:
https://github.com/balena-io/balena-ui/issues/3099#issuecomment-828532247

Don't let this comment of mine be a blocker to anything, but take it into account before making your decision. :-)

Reply:
Hey Paulo, thanks for your comment, After discussion in the May 4th product call, we've agree on keeping the restart behavior as it is, However, the restart behavior throughout the Supervisor is not consistent, so I'll be working on that in another PR. An additional change we're making, which I think you suggested as well, is to clarify in the docs (and on the dashboard actions tab) that restart will remove + recreate containers.